soh/src/code/z_parameter.c

```diff
@@ -2567,15 +2567,43 @@ u16 Randomizer_Item_Give(PlayState* play, GetItemEntry giEntry) {
     }
 
     if (item == RG_PROGRESSIVE_BOMBCHUS) {
-        if (INV_CONTENT(ITEM_BOMBCHU) == ITEM_NONE) {
-            INV_CONTENT(ITEM_BOMBCHU) = ITEM_BOMBCHU;
-            AMMO(ITEM_BOMBCHU) = 20;
-        } else {
-            AMMO(ITEM_BOMBCHU) += AMMO(ITEM_BOMBCHU) < 5 ? 10 : 5;
-            if (AMMO(ITEM_BOMBCHU) > 50) {
-                AMMO(ITEM_BOMBCHU) = 50;
-            }
-        }
```

Comment:
I don't think this section should have been removed either.  If your infinite upgrades are turned off, and you undo the change above to the number of progressive bombchu packs in the pool, then we still need this code to handle getting multiple progressive bombchu packs.

Reply:
This is fine because of the code added in `randomizer.cpp`:
```cpp
case RG_PROGRESSIVE_BOMBCHUS:
    if (INV_CONTENT(ITEM_BOMBCHU) == ITEM_NONE) {
        return (GetItemID)RG_PROGRESSIVE_BOMBCHUS;
    } else if (infiniteUpgrades) {
        return (GetItemID)RG_BOMBCHU_INF;
    } else if (AMMO(ITEM_BOMBCHU) < 5) {
        return GI_BOMBCHUS_10;
    } else {
        return GI_BOMBCHUS_5;
    }
```
Now the progressive bombchus rg is only used when you originally get them, I made this change because it gave me problems the way it was implemented before and I figured it was fine as progressive bombchus are the only item that does logic in `Randomizer_Item_Give`. I can change it back if it's preferred that way.

Reply:
Also as a side effect of this, progressive bombchus only sparkle if you don't have them yet, instead of always.